File: solutions/1586-longest-subarray-of-1s-after-deleting-one-element/solution.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        //sliding window
        int left = 0;
        int maxlength = 0;
        int zeros = 0;
        for(int right=0;right<nums.size();right++){
            if(nums[right]!=1){
                zeros+=1;
            }
            if(zeros>1){
                while(nums[left]!=0){
                    left++;
                }
                left++;
                zeros -= 1;
            }
            maxlength = max(maxlength,right-left);
        }
        return maxlength;
    }
};

File: solutions/1586-longest-subarray-of-1s-after-deleting-one-element/solution.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left = 0
        zeroCount = 0
        maxwindowsize = 0

        for i in range(0,len(nums)):
            zeroCount += (nums[i]==0)
            while(zeroCount>1):
                zeroCount -= (nums[left]==0)
                left+=1
            
            maxwindowsize= max(maxwindowsize,i-left)
        return maxwindowsize
            

        
